### PR TITLE
IGNITE-12981: Fix pme-free snapshot exchange if coordinator left the cluster

### PR DESCRIPTION
**Pull request by Mmuzaf**
Fixed:
- `wasRebalanced` flag doesn't keep after snapshot
- `ExchangeType` must be `NONE` for the snapshot on a client node
- exchange aware component is not covered by test for a snapshot operation

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotSelfTest.java [hunk 1]
```diff
@@ -854,6 +863,190 @@ public void testClusterSnapshotWithSharedCacheGroup() throws Exception {
...
+        assertThrowsAnyCause(log,
+            fut::get,
+            IgniteException.class,
+            "Snapshot creation has been finished with an error");
+
+        assertEquals("Snapshot futures expected: " + exchFuts, 3, exchFuts.size());
+
+        for (T2<GridDhtPartitionExchangeId, Boolean> exch : exchFuts)
+            assertFalse("Snapshot `rebalanced` must be false with moving partitions: " + exch.get1(), exch.get2());
+
```
Comment (NSAmelchev) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotSelfTest.java:989]: Unnecessary line brake

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotSelfTest.java [hunk 2]
```diff
@@ -854,6 +863,190 @@ public void testClusterSnapshotWithSharedCacheGroup() throws Exception {
...
+                    /** {@inheritDoc} */
+                    @Override public void onDoneAfterTopologyUnlock(GridDhtPartitionsExchangeFuture fut) {
+                        assertEquals("Exchange order violated: " + fut.firstEvent(), 3, order.getAndSet(0));
+                    }
+                });
+
+            comps.put(((IgniteEx)ig).localNode().id(), comp);
+        }
+
+        ignite.snapshot().createSnapshot(SNAPSHOT_NAME)
```
Comment (NSAmelchev) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotSelfTest.java:1031]: Can be one line

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotSelfTest.java (hunk 1 above)
Comment (Mmuzaf) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotSelfTest.java:989]: fixed.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotSelfTest.java (hunk 2 above)
Comment (Mmuzaf) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotSelfTest.java]: fixed.

**Pull request merged**